### PR TITLE
Added detection of operative system to the unattended installer

### PR DESCRIPTION
**Pull request by verdx**
|Related issue|
|---|
|https://github.com/wazuh/wazuh-packages/issues/1228|

## Description

Using the tool https://github.com/wazuh/wazuh/blob/v4.3.0-rc3/src/init/dist-detect.sh the unattended installer now detects the operative system and if it is not supported and the option -Fos|--force-os hasn`t been specified it stops the installation.

## Testing
 - [x] Centos
 - [x] Ubuntu
 - [x] Debian
 - [x] Opensuse
 - [x] Fedora
 - [x] Amazon linux
 - [x] RedHat
 - [x] Suse
 - [x] Oracle
 - [x] Rocky
 - [x] Alma

### CODE REVIEW

**Comment by verdx** [comment]
Still have to do the unit tests for this new function in checks.sh

File: unattended_installer/install_functions/checks.sh [hunk 1]
```diff
@@ -6,6 +6,22 @@
 # License (version 2) as published by the FSF - Free Software
 # Foundation.
 
+systems_list=("centos","ubuntu","debian","fedora","amzn","rhel","opensuse-leap", "opensuse-tumbleweed","sles","ol","rocky","almalinux")
```
Comment (alberpilot) [inline unattended_installer/install_functions/checks.sh:9]: This list must be updated with: https://opensearch.org/docs/latest/opensearch/install/compatibility/

**Review by alberpilot — CHANGES_REQUESTED** [review]
Please review the requested changes

File: unattended_installer/install_functions/checks.sh (hunk 1 above)
Comment (verdx) [reply unattended_installer/install_functions/checks.sh]: Done in https://github.com/wazuh/wazuh-packages/pull/1222/commits/32e02871a05c29f90715359ed018dce058aad542

**Pull request closed without merging**